src/legacy/core_plugins/data/public/search/search_bar/components/saved_query_management/saved_query_management_component.tsx

```diff
@@ -61,11 +60,17 @@ export const SavedQueryManagementComponent: FunctionComponent<Props> = ({
 }) => {
   const [isOpen, setIsOpen] = useState(false);
   const [savedQueries, setSavedQueries] = useState([] as SavedQuery[]);
+  const [count, setTotalCount] = useState(0);
   const [activePage, setActivePage] = useState(0);
 
   useEffect(() => {
     const fetchQueries = async () => {
-      const allSavedQueries = await savedQueryService.getAllSavedQueries();
+      const savedQueryCount = await savedQueryService.getSavedQueryCount();
+      setTotalCount(savedQueryCount);
+      const allSavedQueries = await savedQueryService.getAllSavedQueries(
+        savedQueryCount,
+        activePage + 1
+      );
```

Comment:
When we spoke on zoom I was thinking we should only request as many saved queries as we need for the current page. That way we're not requesting more saved objects than we need, in case someone has a very large number of saved queries. When the user requests a new page, we load more results.

Reply:
I started going down that route but realized we'll need to refactor a bit. While the saved queries do update on state, `savedQueryRows` doesn't get called again after a state update. 
I'll give it another go